src/functions/sendDailyMessage/handler.ts

```diff
@@ -16,18 +17,23 @@ import { TimeEntry } from "src/generated/client/api";
  */
 const sendDailyMessage: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async (event: ValidatedAPIGatewayProxyEvent<typeof schema>) => {
   try {
+    let yesterday = DateTime.now().minus({ days: 1 }).toISODate();
+
     const timebankUsers = await TimeBankApiProvider.getTimebankUsers();
     const slackUsers = await SlackApiUtilities.getSlackUsers();
+    const timeRegistrations = await ForecastApiUtilities.getTimeRegistrations(yesterday);
 
     const timeEntries: TimeEntry[] = [];
-    const yesterday = DateTime.now().minus({ days: 1 }).toISODate();
+
+    const dayOfWeek = new Date().getDay();
+    (dayOfWeek === 1)? yesterday = DateTime.now().minus({ days: 3 }).toISODate(): yesterday = DateTime.now().minus({ days: 1 }).toISODate();
```

Comment:
Shouldn't this be before the date value is used in await ForecastApiUtilities.getTimeRegistrations(yesterday);?

Also this logic should be moved to TimeUtilities file

Reply:
Done